Lib/test/test_concurrent_futures/executor.py

```diff
@@ -71,6 +73,50 @@ def test_map_timeout(self):
 
         self.assertEqual([None, None], results)
 
+    def test_map_buffersize_validation(self):
+        with self.assertRaisesRegex(
+            ValueError,
+            "buffersize must be None or > 0",
+        ):
+            self.executor.map(str, range(4), buffersize=0)
+
+    def test_map_buffersize(self):
+        ints = range(4)
+        for buffersize in (1, 2, len(ints), len(ints) * 2):
+            with self.subTest(buffersize=buffersize):
+                res = self.executor.map(str, ints, buffersize=buffersize)
+                self.assertEqual(list(res), ["0", "1", "2", "3"])
```

Comment:
Can you use `assertListEqual` instead? if there's an issue, the diff would be nicer. Ditto for the rest.

Reply:
ow good catch 🙏🏻 !